payment/controllers.go

```diff
@@ -779,6 +766,62 @@ func VerifyCredential(service *Service) handlers.AppHandler {
 			}
 
 			return handlers.RenderContent(r.Context(), "Credentials successfully verified", w, http.StatusOK)
+		} else if req.Type == "time-limited" {
+			// Presentation includes a token and token metadata test test
+			type Presentation struct {
+				Issuer    string `json:"issuer"`
+				IssuedAt  string `json:"issuedAt"`
+				ExpiresAt string `json:"expiresAt"`
+				Token     string `json:"token"`
+			}
+
+			var bytes []byte
+			bytes, err = base64.StdEncoding.DecodeString(req.Presentation)
+			if err != nil {
+				return handlers.WrapError(err, "Error in decoding presentation", http.StatusBadRequest)
+			}
+
+			var presentation Presentation
+			err = json.Unmarshal(bytes, &presentation)
+			if err != nil {
+				return handlers.WrapError(err, "Error in presentation formatting", http.StatusBadRequest)
+			}
+
+			// Ensure that the credential being redeemed (opaque to merchant) matches the outer credential details
+			issuerID, err := encodeIssuerID(req.MerchantID, req.SKU)
+			if err != nil {
+				return handlers.WrapError(err, "Error in outer merchantId or sku", http.StatusBadRequest)
+			}
+			if issuerID != presentation.Issuer {
+				return handlers.WrapError(nil, "Error, outer merchant and sku don't match issuer", http.StatusBadRequest)
+			}
+
+			timeLimitedSecret := cryptography.NewTimeLimitedSecret([]byte(os.Getenv("BRAVE_MERCHANT_KEY")))
+
+			issuedAt, err := time.Parse("2006-01-02", presentation.IssuedAt)
+			if err != nil {
+				return handlers.WrapError(err, "Error parsing issuedAt", http.StatusBadRequest)
+			}
+			expiresAt, err := time.Parse("2006-01-02", presentation.ExpiresAt)
+			if err != nil {
+				return handlers.WrapError(err, "Error parsing expiresAt", http.StatusBadRequest)
+			}
+
+			verified, err := timeLimitedSecret.Verify([]byte(presentation.Issuer), issuedAt, expiresAt, presentation.Token)
+			if err != nil {
+				return handlers.WrapError(err, "Error in token verification", http.StatusBadRequest)
+			}
+
+			if verified {
+				// check against expiration time
+				if time.Now().After(expiresAt) {
+					return handlers.RenderContent(r.Context(), "Credentials expired", w, http.StatusBadRequest)
```

Comment:
Shouldn't this be a 403 (`http.StatusForbidden`)?

Reply:
fixed: https://github.com/brave-intl/bat-go/pull/863/commits/3ee9f1c65a4b7df7629e9b69ac397d693d689eb0